crypto/signatures_test.go

```diff
@@ -276,3 +277,67 @@ func TestIntegrationSigKeyGeneration(t *testing.T) {
 		t.Error("corruption failed")
 	}
 }
+
+// TestReadWriteSignedObject tests the WriteSignedObject and ReadSignedObject
+// functions, which are inverses of each other.
+func TestReadWriteSignedObject(t *testing.T) {
+	sk, pk, err := GenerateKeyPair()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// write signed object
+	b := new(bytes.Buffer)
+	if err = WriteSignedData(b, "foo", sk); err != nil {
+		t.Fatal(err)
+	}
+
+	// read signed object
+	var read string
+	err = ReadSignedData(b, &read, 11, pk)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if read != "foo" {
+		t.Fatal("encode/decode mismatch: expected 'foo', got", read)
+	}
+
+	// corrupt signature
+	if err = WriteSignedData(b, "foo", sk); err != nil {
+		t.Fatal(err)
+	}
+	buf := b.Bytes()
+	_, err = rand.Read(buf[11:])
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = ReadSignedData(b, &read, 11, pk)
+	if err != errInvalidSignature {
+		t.Fatal("expected errInvalidSignature, got", err)
+	}
+
+	// not enough object data
+	err = ReadSignedData(b, &read, 11, pk)
+	if err != io.EOF {
+		t.Fatal("expected EOF, got", err)
+	}
+	// not enough signature data
+	if err := encoding.WriteObject(b, "foo"); err != nil {
+		t.Fatal(err)
+	}
+	err = ReadSignedData(b, &read, 11, pk)
+	if err == nil || err.Error() != "could not decode type crypto.Signature: EOF" {
+		t.Fatal("expected decode error, got", err)
+	}
+}
+
+// TestPublicKey tests the PublicKey method
+func TestPublicKey(t *testing.T) {
+	sk, pk, err := GenerateKeyPair()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if sk.PublicKey() != pk {
+		t.Fatal("PublicKey does not match actual public key:", pk, sk.PublicKey())
+	}
+}
```

Comment:
That would be testing the functionality of the public key, which doesn't really have anything to do with the function being tested.
All `PublicKey()` does is take a secret key and return its corresponding public key. As long as the result matches the previously-generated `pk`, `PublicKey()` has done its job.
Nothing wrong with generating 5 or 10 keys rather than just one, but since it's random and it gets run on every build, it doesn't seem much more thorough to me.